### PR TITLE
Addition of PEL message method for dump delete/offload

### PR DESCRIPTION
**Pull request by Swarnendu-R-C**
Addition of PEL message method for dump delete/offload

Adding a createPEL method which will log PEL messages.
This method has PEL severity as Informational by default
unless mentioned otherwise

Gerrit review link https://gerrit.openbmc.org/c/openbmc/phosphor-debug-collector/+/53547

Create an info PEL when doing a dump invalidation/delete
    
When a dump is been ofloaded and subsequently deleted by a HMC
managed system or by any other means now it is logging
a PEL message having the information of about that deleted
dump file like Dump Type, Dump File Name and Dump ID.
image (1)
image

Signed-off-by: swarnendu.roy.chowdhury@ibm.com

### CODE REVIEW

File: dump_utils.cpp [hunk 1]
```diff
@@ -139,5 +139,68 @@ bool isHostQuiesced()
...
+        // Create a SystemD D-Bus object to be used in the co-routine
+        const auto systemd = sdbusplus::async::proxy()
+                                 .service(loggerService)
+                                 .path(loggerObjectPath)
+                                 .interface(loggerCreateInterface)
+                                 .preserve();
+
+        log<level::INFO>("Going for writing PEL via co-routine");
+        co_await systemd.call<>(ctx, "Create", errIntf, pelSev, userDataMap);
+        log<level::INFO>("Writing PEL via co-routine is done now");
```
Comment (devenrao) [inline dump_utils.cpp:192]: remove

**Comment by devenrao** [comment]
> Addition of PEL message method for dump delete/offload
> 
> Adding a createPEL method which will log PEL messages. This method has PEL severity as Informational by default unless mentioned otherwise
> 
> Change-Id-1: [5b117f5](https://github.com/ibm-openbmc/phosphor-debug-collector/commit/5b117f5b53e4c188474bea022a82ea13f426b459) Change-Id-2: [15b47b9](https://github.com/ibm-openbmc/phosphor-debug-collector/commit/15b47b977cf48c2c09dbd12155c16ae35eba5d73)
> 
> Signed-off-by: [swarnendu.roy.chowdhury@ibm.com](mailto:swarnendu.roy.chowdhury@ibm.com)

no need of Change-Id

**Review by devenrao — COMMENTED** [review]
please capture test results

File: dump_utils.cpp (hunk 1 above)
Comment (Swarnendu-R-C) [reply dump_utils.cpp]: Removed

File: dump_utils.cpp [hunk 2]
```diff
@@ -139,5 +139,68 @@ bool isHostQuiesced()
...
+        constexpr auto loggerService = "xyz.openbmc_project.Logging";
+
+        // Create a SystemD D-Bus object to be used in the co-routine
+        const auto systemd = sdbusplus::async::proxy()
+                                 .service(loggerService)
+                                 .path(loggerObjectPath)
+                                 .interface(loggerCreateInterface)
+                                 .preserve();
+
+        log<level::INFO>("Going for writing PEL via co-routine");
```
[the remark this answers is not in the document]
Comment (Swarnendu-R-C) [reply dump_utils.cpp]: Removed

**Comment by Swarnendu-R-C** [comment]
> please capture test results

Captured now

File: dump_utils.cpp [hunk 3]
```diff
@@ -139,5 +141,59 @@ bool isHostQuiesced()
...
+        constexpr auto dumpFileTypeString = "Dump Type";
+        constexpr auto dumpIdString = "Dump ID";
+
+        const std::unordered_map<std::string_view, std::string_view>
+            userDataMap = {{dumpIdString, std::to_string(dumpId)},
+                           {dumpFileString, dumpFilePath},
+                           {dumpFileTypeString, dumpFileType}};
+
+        // Set up a connection to D-Bus object
+        auto dBus = sdbusplus::bus::new_default();
```
Comment (devenrao) [inline dump_utils.cpp:166]: pass bus from the caller

File: dump_utils.hpp [hunk 4]
```diff
@@ -10,7 +10,6 @@
 #include <phosphor-logging/elog-errors.hpp>
 #include <phosphor-logging/elog.hpp>
 #include <phosphor-logging/log.hpp>
-#include <sdbusplus/bus.hpp>
```
Comment (devenrao) [inline dump_utils.hpp:13]: why is this removed not related to your change

File: dump_utils.cpp [hunk 5]
```diff
@@ -139,5 +141,59 @@ bool isHostQuiesced()
...
+        auto busMethod = dBus.new_method_call(loggerService, loggerObjectPath,
+                                              loggerCreateInterface, "Create");
+        busMethod.append(errIntf, pelSev, userDataMap);
+
+        // Implies this is a call from Manager. Hence we need to make an async
+        // call to avoid deadlock with Phosphor-logging.
+        auto slot =
```
Comment (devenrao) [inline dump_utils.cpp:173]: did you ran clang?

Comment (devenrao) [inline dump_utils.cpp:173]: what is slot here? ret would be more apt

File: bmcstored_dump_entry.cpp [hunk 6]
```diff
@@ -33,6 +36,14 @@ void Entry::delete_()
...
     // Remove Dump entry D-bus object
     phosphor::dump::Entry::delete_();
+
+    // Log PEL for dump delete/offload
+    log<level::INFO>("Log PEL for dump delete or offload");
+
+    phosphor::dump::createPEL(
+        strDumpFileName, "BMC Dump", id,
+        "xyz.openbmc_project.Logging.Entry.Level.Informational",
+        "xyz.openbmc_project.Dump.Error.Invalidate");
```
Comment (devenrao) [inline bmcstored_dump_entry.cpp:46]: u can do it before delete that way no need to create strDumpFileName variable, nothing serious if this succeeds and delete fails, delete fail is a core dump

File: dump-extensions/openpower-dumps/resource_dump_entry.cpp [hunk 7]
```diff
@@ -82,6 +83,14 @@ void Entry::delete_()
 
     // Remove Dump entry D-bus object
     phosphor::dump::Entry::delete_();
+
+    // Log PEL for dump /offload
+    log<level::INFO>("Log PEL for dump delete or offload");
```
Comment (devenrao) [inline dump-extensions/openpower-dumps/resource_dump_entry.cpp:88]: remove trace not essential

**Comment by ojayanth** [comment]
@devenrao please post the reveiw comments in the gerrit link. 
@Swarnendu-R-C please include Gerrit reveiw link.

File: dump_utils.hpp (hunk 4 above)
Comment (Swarnendu-R-C) [reply dump_utils.hpp:13]: Actually I realized that it was an unnecessary inclusion thus removed. But re-included on your suggestion

File: dump_utils.cpp (hunk 5 above)
Comment (Swarnendu-R-C) [reply dump_utils.cpp:173]: Changed

Comment (Swarnendu-R-C) [reply dump_utils.cpp:173]: Yes, for each cpp/hpp files

File: dump_utils.cpp (hunk 3 above)
Comment (Swarnendu-R-C) [reply dump_utils.cpp:166]: Done

File: bmcstored_dump_entry.cpp (hunk 6 above)
Comment (Swarnendu-R-C) [reply bmcstored_dump_entry.cpp:46]: Moved before delete

File: dump-extensions/openpower-dumps/resource_dump_entry.cpp (hunk 7 above)
Comment (Swarnendu-R-C) [reply dump-extensions/openpower-dumps/resource_dump_entry.cpp:88]: Removed

File: dump-extensions/openpower-dumps/system_dump_entry.cpp [hunk 8]
```diff
@@ -85,6 +86,14 @@ void Entry::delete_()
 
     // Remove Dump entry D-bus object
     phosphor::dump::Entry::delete_();
+
+    // Log PEL for dump delete/offload
+    log<level::INFO>("Log PEL for dump delete or offload");
```
[the remark this answers is not in the document]
Comment (Swarnendu-R-C) [reply dump-extensions/openpower-dumps/system_dump_entry.cpp:91]: Removed

**Comment by Swarnendu-R-C** [comment]
> @devenrao please post the reveiw comments in the gerrit link. @Swarnendu-R-C please include Gerrit reveiw link.

@ojayanth , Gerrit link is included now with the latest changes

**Review by devenrao — APPROVED** [review]
